### PR TITLE
*: rename test scripts and update test pod readme

### PR DESCRIPTION
**Pull request by hasbro17**
Renamed:
- ci/run2 to ci/run_e2e
- test-pod.yaml to test-pod-tmpl.yaml

Also updated the README for `test/pod`.

### CODE REVIEW

File: test/pod/README.md [hunk 1]
```diff
@@ -11,24 +11,42 @@ The e2e tests need access to an S3 bucket for testing. Create a secret containin
...
 
 ```sh
-TEST_IMAGE=gcr.io/coreos-k8s-scale-testing/etcd-operator-e2e \
   TEST_NAMESPACE=e2e \
-  OPERATOR_IMAGE=quay.io/coreos/etcd-operator:dev \
-  TEST_S3_BUCKET=my-bucket \
-  TEST_AWS_SECRET=aws-secret \
-  test/pod/run-test-pod
+    POD_NAME=testing-e2e \
+    TEST_POD_SPEC=test-pod-spec.yaml \
```
Comment (hongchaodeng) [inline test/pod/README.md]:
```
TEST_POD_SPEC=${PWD}/test-pod-spec.yaml
```

File: test/pod/README.md [hunk 2]
```diff
@@ -11,24 +11,42 @@ The e2e tests need access to an S3 bucket for testing. Create a secret containin
...
+    KUBECONFIG=/path/to/kubeconfig \
+    test/pod/run-test-pod
 ```
 
+The logfiles for all the pods in the test namespace will be saved at `${REPO_ROOT}/_output/logs`.
```
Comment (hongchaodeng) [inline test/pod/README.md]: REPO_ROOT -> CWD

File: hack/ci/run_e2e [hunk 3]
```diff
@@ -15,47 +15,42 @@ kubectl version
...
   gcloud docker -a
   hack/build/build
   IMAGE=${OPERATOR_IMAGE} hack/build/docker_push
 fi
 
 BUILD_E2E=${BUILD_E2E:-true}
 if [[ ${BUILD_E2E} == "true" ]]; then
+  TEST_IMAGE=${TEST_IMAGE:-"gcr.io/coreos-k8s-scale-testing/etcd-operator-e2e:${GIT_VERSION}"}
   gcloud docker -a
   hack/build/e2e/docker_push
```
Comment (hongchaodeng) [inline hack/ci/run_e2e]: Should be
```
TEST_IMAGE=... \
  hack/build/e2e/docker_push
```

**Comment by hasbro17** [comment]
@hongchaodeng PTAL

**Comment by hongchaodeng** [comment]
LGTM

**Pull request merged**